pkg/alertmanager/amcfg.go

```diff
@@ -127,6 +129,52 @@ func (cg *configGenerator) generateConfig(
 	return yaml.Marshal(baseConfig)
 }
 
+// sanitizeBaseConfig checks provided config is valid for the version it is being
+// generated against - strips fields from config is unsupported
+func (cg *configGenerator) sanitizeBaseConfig(amConf *alertmanagerConfig) {
+	if amConf == nil {
+		return
+	}
+	// as of v0.22 AlertManager config supports passing URL via file name
+	// todo(pgough) - remove when operator supports >= 0.22 only
+	fileURLAllowed := cg.amVersion.GTE(semver.MustParse("0.22.0"))
+	if amConf.Global != nil {
+		if amConf.Global.SlackAPIURL != nil && amConf.Global.SlackAPIURLFile != "" {
+			msg := "slack_api_url and slack_api_url_file are mutually exclusive - slack_api_url has been preferred"
+			level.Warn(cg.logger).Log("msg", msg)
+			amConf.Global.SlackAPIURLFile = ""
+		}
+
+		if amConf.Global.SlackAPIURLFile != "" && !fileURLAllowed {
+			msg := "slack_api_url_file supported in AlertManager >= 0.22 only - dropping field from provided config"
+			level.Warn(cg.logger).Log("msg", msg, "current_version", cg.amVersion.String())
+			amConf.Global.SlackAPIURLFile = ""
+		}
+	}
+
+	if amConf.Receivers == nil {
+		return
+	}
+
```

Comment:
It would work equally since a nil slice is equivalent to an empty slice for `range`

```suggestion
```

Reply:
indeed :)